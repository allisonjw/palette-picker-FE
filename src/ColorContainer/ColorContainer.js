import ColorCard from '../ColorCard/ColorCard';
import React, { useState } from 'react';
import './ColorContainer.scss';

const ColorContainer = () => {
    const [initialPalette, setInitialPalette] = useState([
        {palette_id: 1, color: '#CA58AD', locked: false},
        {palette_id: 2, color: '#40F8A5', locked: false},
        {palette_id: 3, color: '#7D06F0', locked: false},
        {palette_id: 4, color: '#A545F5', locked: false},
        {palette_id: 5, color: '#341ADB', locked: false}
    ]);     

      const changeColor = () => {
        const randomColor = initialPalette.map(initPalette => {
            if (initPalette.locked === false) {
                initPalette.color = `${Math.floor(Math.random()*16777215).toString(16)}`;
            }
            return initPalette
        })
        setInitialPalette(randomColor)
      }

      const lockColor = (id) => {
          const updatedColors = initialPalette.map(color => {
            if (color.palette_id === id) {
              color.locked = !color.locked
            }
            return color;
          });
          console.log('lockcolor:', updatedColors)
          setInitialPalette(updatedColors)
      }
      
    //   const generateRandomColor = () => {
    //       return Math.floor(Math.random()*16777215).toString(16);
    //   }

      const displayColorCard = initialPalette.map(palette => {
        return (
            <ColorCard 
              {...palette}
              key={palette.palette_id}
              lockColor={lockColor}
            //   changeColor={changeColor}
            />
        )
      })

      
    return(
        <>
        <section className="section_colorContainer">
            {displayColorCard}
            {/* <ColorCard 
                // onClick={reply_click(id)}
                color={initialPalette[0].color} 
                initialPalette={initialPalette} 
                locked={initialPalette[0].locked} 
                id={initialPalette[0].palette_id}
                lockColor={lockColor}/>
            <ColorCard 
                color={initialPalette[1].color} 
                initialPalette={initialPalette} 
                locked={initialPalette[1].locked}
                id={initialPalette[1].palette_id} 
                lockColor={lockColor}/>
            <ColorCard 
                color={initialPalette[2].color} 
                initialPalette={initialPalette} 
                locked={initialPalette[2].locked} 
                id={initialPalette[2].palette_id}
                lockColor={lockColor}/>
            <ColorCard 
                color={initialPalette[3].color} 
                initialPalette={initialPalette} 
                locked={initialPalette[3].locked} 
                id={initialPalette[3].palette_id}
                lockColor={lockColor}/>
            <ColorCard 
                color={initialPalette[4].color} 
                initialPalette={initialPalette} 
                locked={initialPalette[4].locked}
                id={initialPalette[4].palette_id} 
                lockColor={lockColor}/> */}
        </section>
        <section className="generate_btn-section">   
        <button
            type='button'
            className='generate_colors-btn'
            onClick={() => changeColor()}>GENERATE COLORS
        </button>
        </section>
        </>
    )
}

export default ColorContainer;