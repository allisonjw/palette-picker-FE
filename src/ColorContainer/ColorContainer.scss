.section_colorContainer {
    display: flex;
    height: 30vh;
    box-shadow: inset 0px 20px 16px 5px #090e0f;
    padding: 3px;
}

.generate_btn-section {
    display: flex;
    width: 100%;
    justify-content: center;
}

.generate_colors-btn {
  background-color: black;
    border-radius: 6px;
    color: white;
    height: 60px;
    font-size: 1.3em;
    font-family: 'Shadows Into Light', cursive;
    outline: none;
    margin-top: .8%;
      &:hover {
        cursor: pointer;
        transform: scale(1.1);
        transition: all .4s ease-in-out;
      };
    
}