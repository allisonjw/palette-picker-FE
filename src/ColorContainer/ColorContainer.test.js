import React from 'react';
import { render } from '@testing-library/react';
import ColorContainer from './ColorContainer';
import { shallow } from 'enzyme';

describe('ColorContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ColorContainer />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});