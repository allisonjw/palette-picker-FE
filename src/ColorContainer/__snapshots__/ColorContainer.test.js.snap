// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorContainer should match the snapshot 1`] = `
<Fragment>
  <section
    className="section_colorContainer"
  >
    <ColorCard
      color=""
      key="1"
      lockColor={[Function]}
      locked={false}
      palette_id={1}
    />
    <ColorCard
      color=""
      key="2"
      lockColor={[Function]}
      locked={false}
      palette_id={2}
    />
    <ColorCard
      color=""
      key="3"
      lockColor={[Function]}
      locked={false}
      palette_id={3}
    />
    <ColorCard
      color=""
      key="4"
      lockColor={[Function]}
      locked={false}
      palette_id={4}
    />
    <ColorCard
      color=""
      key="5"
      lockColor={[Function]}
      locked={false}
      palette_id={5}
    />
  </section>
  <section
    className="generate_btn-section"
  >
    <button
      className="generate_colors-btn"
      onClick={[Function]}
      type="button"
    >
      GENERATE COLORS
    </button>
  </section>
  <PaletteForm
    initialPalette={
      Array [
        Object {
          "color": "",
          "locked": false,
          "palette_id": 1,
        },
        Object {
          "color": "",
          "locked": false,
          "palette_id": 2,
        },
        Object {
          "color": "",
          "locked": false,
          "palette_id": 3,
        },
        Object {
          "color": "",
          "locked": false,
          "palette_id": 4,
        },
        Object {
          "color": "",
          "locked": false,
          "palette_id": 5,
        },
      ]
    }
  />
</Fragment>
`;
